multinet/api.py

```diff
@@ -24,11 +27,62 @@
     AlreadyExists,
     RequiredParamsMissing,
 )
-from multinet.user import current_user
+from multinet.user import current_user, find_user_from_id
 
 bp = Blueprint("multinet", __name__)
 
 
+# Included here due to circular imports
+# TODO: Remove once implementing new ORM and permission storage
+def _permissions_id_to_user(permissons: WorkspacePermissions) -> Dict:
+    """Transform permission documents to directly container user info."""
+
+    new_permissions = deepcopy(permissons)
+    for role, users in new_permissions.items():
+        if role == "public":
+            continue
+
+        if role == "owner":
+            assert isinstance(users, str)
+
+            user = find_user_from_id(users)
+            if user is not None:
+                # Not sure why mypy complains about `user` here
+                new_permissions["owner"] = asdict(user)  # type: ignore
+        else:
+            assert isinstance(users, list)
+
+            new_users = []
+            for sub in users:
+                user = find_user_from_id(sub)
+                if user is not None:
+                    new_users.append(asdict(user))
+
+            # Ignoring requirement of literal string keys
+            new_permissions[role] = new_users  # type: ignore
+
+    return cast(Dict, new_permissions)
```

Comment:
Is `WorkspacePermissions` a dataclass? How does this `cast` operator work to turn that into a `Dict`?

Reply:
It's a `TypedDict`, and I'm not sure how the `cast` operator works internally, but it's used here to remove the restrictions of it being a `WorkspacePermissions` dict, because it isn't anymore, due to the replacements above. I'm actually about to push a change where I move this operation, but it'll still be in this function.